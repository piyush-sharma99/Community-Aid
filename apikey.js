/*
 *  ClassName: apikey.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=8zT6CYu0iYQ
 *
 */

export const firebaseConfig = {
  apiKey: "AIzaSyDGliQfMfKpXUllG5ZPyClJL6C8VgFKXo8",
  authDomain: "community-aid.firebaseapp.com",
  databaseURL:
    "https://community-aid-default-rtdb.europe-west1.firebasedatabase.app",
  projectId: "community-aid",
  storageBucket: "community-aid.appspot.com",
  messagingSenderId: "529896701339",
  appId: "1:529896701339:web:1c30369d39f8654b01353e",
  measurementId: "G-PPD28P0PC6",
};
