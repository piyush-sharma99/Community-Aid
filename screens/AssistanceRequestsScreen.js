import React from 'react';
import { StyleSheet, Text, View, TextInput, Button } from 'react-native';

const AssistanceRequestScreen = props => {
    return(
        <View style={styles.screen}>
            <Text>Make Assistance Request</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    screen: {
        flex:1,
        justifyContent: 'center',
        alignContent: 'center'
    }
});

export default AssistanceRequestScreen;