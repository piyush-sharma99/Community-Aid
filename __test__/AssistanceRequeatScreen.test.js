import React from "react";
import AssistanceRequestsScreen from "../screens/AssistanceRequestsScreen";
import { firebaseConfig } from "../apikey";
import { cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

afterEach(cleanup);

test("Assistance Request renders correctly", () => {
  const tree = renderer.create(<AssistanceRequestsScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
