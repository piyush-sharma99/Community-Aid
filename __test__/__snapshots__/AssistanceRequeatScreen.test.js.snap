// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assistance Request renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexWrap": "wrap",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#fb5b5a",
          "borderRadius": 15,
          "borderWidth": 3,
          "color": "white",
          "fontSize": 30,
          "fontWeight": "bold",
          "marginBottom": 40,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 30,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      Choose one below:
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fb5b5a",
          "borderRadius": 25,
          "height": 230,
          "justifyContent": "center",
          "margin": 10,
          "marginTop": 15,
          "opacity": 1,
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 1,
          "width": "95%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#fb5b5a",
            "borderColor": "white",
            "borderRadius": 25,
            "borderWidth": 3,
            "elevation": 10,
            "height": "100%",
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 3,
              "width": 3,
            },
            "shadowOpacity": 1,
            "shadowRadius": 10,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexGrow": 1,
              "flexShrink": 1,
            }
          }
        >
          <View
            index={0}
            siblings={
              Array [
                "View",
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fb5b5a",
                "borderRadius": 25,
                "flex": 1,
                "justifyContent": "center",
              }
            }
            total={1}
          >
            <Text />
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
            >
               Make Assistance Request
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fb5b5a",
          "borderRadius": 25,
          "height": 230,
          "justifyContent": "center",
          "margin": 10,
          "marginTop": 15,
          "opacity": 1,
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 1,
          "width": "95%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#fb5b5a",
            "borderColor": "white",
            "borderRadius": 25,
            "borderWidth": 3,
            "elevation": 10,
            "height": "100%",
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 3,
              "width": 3,
            },
            "shadowOpacity": 1,
            "shadowRadius": 10,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexGrow": 1,
              "flexShrink": 1,
            }
          }
        >
          <View
            index={0}
            siblings={
              Array [
                "View",
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fb5b5a",
                "borderRadius": 25,
                "flex": 1,
                "justifyContent": "center",
              }
            }
            total={1}
          >
            <Text />
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
            >
              Pick Up Assistance Requests
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
